### DISCUSSION TITLE
dirtyFields

### DISCUSSION THREAD

**Discussion by kceron** [react-hook-form/react-hook-form#5336] [Q&A]
Hi, I have the following code. I wonder what am I doing wrong? When I submit the form, console.log(data) gets the right submitted values but when I try using dirtyFields to submit only updated values, all of them are turn to **true**

const { handleSubmit, register, setValue, control, formState, reset } =
    useForm({
      defaultValues,
    });
  const onSubmit = (data) => {
    console.log("DIRTY FIELDS", formState);
    console.log(data);
    const filteredVals = formState.dirtyFields;
    console.log("filtered: ", filteredVals);
    reset({ defaultValues });
  };

// input is in different file

<input
        name={name}
        label=""
        className={classes.root}
        defaultValue={value}
        onChange={(e) => {
          console.log("ON CHANGE VALUE", e.target.value),
            setValue(
              input.name,
              e.target.value,
              {
                shouldDirty: true,
              }
              // handleInputChange(inputElementRef)
            );
        }}
      ></input>

**Comment by bluebill1049** [comment]
can you provide a CSB.

**Reply by kceron** [reply]
I am not sure I can replicate the code in CSB because I have a deeply nested form but basically when I assign:
const filteredVals = formState.dirtyFields; 
all of the values are turned to **true** , instead of the actual value, when I check _data_ in the submit function, it contains the correct values, and the way I have it in the input field is this: 
<input
name={name}
label=""
className={classes.root}
defaultValue={value}
onChange={(e) => {
console.log("ON CHANGE VALUE", e.target.value),
setValue(
input.name,
e.target.value,
{
shouldDirty: true,
}
);
}}
>
(e.target.value also contains the correct value in onChange)

**Reply by kceron** [reply]
any idea of why might this be happening is appreciate it

**Reply by thanh-nguyen-95** [reply]
This is unrelated to the discussion but I think it's an important note.

Since you cannot provide a CodeSandbox, please use the syntax below next time you want to include a code snippet. It improves readability a lot.

\```jsx
your code here
\```

An example:

```jsx
<input
  name={name}
  label=""
  className={classes.root}
  defaultValue={value}
  onChange={(e) => {
    console.log("ON CHANGE VALUE", e.target.value),
    setValue(
      input.name,
      e.target.value,
      {
        shouldDirty: true,
      }
    );
  }}
/>
```

Just curious, how did you input the values intro your form? Did you use `tab` to switch focus and input some of the values or use your mouse to focus and then input the values?
